scripts/start-test-environment.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/sh
+
+# Try to determine script location
+if [ -n "${BASH_SOURCE:-}" ]; then
+    # Bash-specific path resolution
+    SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+    PROJECT_ROOT="$(dirname "$SCRIPT_DIR")"
+else
+    # POSIX fallback - assume we're in the repository root
+    PROJECT_ROOT="$(pwd)"
+    SCRIPT_DIR="$PROJECT_ROOT/scripts"
+fi
+
+# Detect if we're being sourced (POSIX-compatible way)
+sourced=0
+if [ -n "$ZSH_EVAL_CONTEXT" ]; then 
+    case $ZSH_EVAL_CONTEXT in *:file:*) sourced=1;; esac
+elif [ -n "$KSH_VERSION" ]; then
+    [ "$(cd $(dirname -- $0) && pwd -P)/$(basename -- $0)" != "$(cd $(dirname -- ${.sh.file}) && pwd -P)/$(basename -- ${.sh.file})" ] && sourced=1
+elif [ -n "$BASH_VERSION" ]; then
+    (return 0 2>/dev/null) && sourced=1
+else
+    # POSIX fallback - check if we can modify our environment
+    # Try to modify a test variable
+    TEST_VAR="test"
+    if [ -z "${TEST_VAR:-}" ]; then
+        sourced=0
+    else
+        sourced=1
+    fi
+fi
+
+if [ $sourced -eq 0 ]; then
+    echo "Warning: This script should be sourced to set environment variables"
+    echo "Please either:"
+    echo "1. Source this script:  source scripts/start-test-environment.sh"
+    echo "2. Or manually set these environment variables:"
+    echo "   export REQUESTS_TEST_HOST_HTTP=localhost:8080"
+    echo "   export REQUESTS_HTTP_PROXY=localhost:9002"
+    echo "   export REQUESTS_HTTP_PROXY_AUTH=localhost:9003"
+    echo "   export REQUESTS_HTTP_PROXY_AUTH_USER=test"
+    echo "   export REQUESTS_HTTP_PROXY_AUTH_PASS=pass"
+    if [ -n "${BASH_VERSION:-}" ]; then
+        exit 1
+    fi
+fi
+
+PID_DIR="${PROJECT_ROOT}/tests/utils/pids"
+
+# Check if mitmproxy is installed
+if ! command -v mitmdump >/dev/null 2>&1; then
+    echo "Error: mitmproxy is not installed. Please install it with: pip3 install mitmproxy"
```

Comment:
Should there be a check first to see if Python/Pip3 is installed ?

Reply:
I'd consider that out of scope. If you can get mitmproxy to run without having pip around or python properly set up, we should'nt break. We only need mitmproxy to work _somehow_.